### PR TITLE
Feat: pipfile handler check & share common functionality

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?
- drop micromatch it was under used and not needed, the check is much simpler here as CLI already had done the work of classifying the project type as python.
- add a new check with tests for a project that has a `Pipfile` manifest  + tests (nothing is using it yet)

### CODE REVIEW

File: packages/snyk-fix/src/plugins/python/get-handler-type.ts [hunk 1]
```diff
@@ -18,11 +16,9 @@ export function getHandlerType(
...
 
 export function isRequirementsTxtManifest(targetFile: string): boolean {
-  return micromatch.isMatch(
-    targetFile,
-    // micromatch needs **/* to match filenames that may include folders
-    ['*.txt'].map(
-      (f) => '**/' + f,
-    ),
-  );
+  return targetFile.endsWith('.txt');
```
Comment (aarlaud) [inline packages/snyk-fix/src/plugins/python/get-handler-type.ts:19]: does that handle the folder situation mentioned on the left ?

File: packages/snyk-fix/src/plugins/python/handlers/valdidate-required-data.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { MissingRemediationDataError } from '../../../lib/errors/missing-remediation-data';
```
Comment (aarlaud) [inline packages/snyk-fix/src/plugins/python/handlers/valdidate-required-data.ts:1]: typo in filename

**Review by aarlaud — APPROVED** [review]

File: packages/snyk-fix/src/plugins/python/handlers/valdidate-required-data.ts (hunk 2 above)
Comment (lili2311) [reply packages/snyk-fix/src/plugins/python/handlers/valdidate-required-data.ts:1]: which one is the typo? Not quite sure

File: packages/snyk-fix/src/plugins/python/get-handler-type.ts (hunk 1 above)
Comment (lili2311) [reply packages/snyk-fix/src/plugins/python/get-handler-type.ts:19]: adding a test for that use-case, will be supported yep!

**Pull request merged**